### DISCUSSION TITLE
The extracted text get reversed after is interacted with rect

### DISCUSSION THREAD

**Discussion by vinniec2** [pymupdf/PyMuPDF#3590] [Looking for help] [answered]
I tried two ways to extract a single piece of text from a pdf page with tables:

1)removing the boundaryboxes of the tables and then using the apply_redactions() method and then get_text() to get the text.
2)using get_textbox() on a rectangle in the area outside the tables.
Both methods work but return the text in reverse order! from the bottom row up.

If instead I use get_text() without apply_redaction() the text is returned in the right order, but includes unwanted tab text.

Do I have to reorder the text myself with splitlines() and reverse()?

**Comment by JorjMcKie** [comment, marked as answer]
Why don't you simply extract via `page.get_text(sort=True)`?

**Reply by vinniec2** [reply]
It seems to work for page.get_text() but does not seem to work for page.get_textbox(rect) which does not seem to have the sort parameter.

However I looked at the parameters of get_text and it also has the clip parameter which seems to do exactly what get_textbox does but also with the sort parameter.
So can I use page.get_text(clip=rect, sort=True) without any problems instead of get_textbox(rect) ?

**Comment by vinniec2** [comment]
I always used page.get_text(clip=rect, sort=True) and it worked as intended!
thanks
